solidity/test/TestGasPriceOracle.js

```diff
@@ -0,0 +1,175 @@
+const expect = chai.expect
+const assert = chai.assert
+
+describe("GasPriceOracle", () => {
+
+    const owner = accounts[0]
+    const thirdParty = accounts[2]
+
+    let oracle
+
+    before(async () => {
+        oracle = await GasPriceOracle.new({from: owner})
+    })
+
+    beforeEach(async () => {
+        await createSnapshot()
+    })
+
+    afterEach(async () => {
+        await restoreSnapshot()
+    })
+
+    describe("when updating gas price", async () => {
+        it("does not let third party to begin the process", async () => {
+            await expectRevert(
+                oracle.beginGasPriceUpdate(123, {from: thirdParty}),
+                "caller is not the owner"
+            )
+        })
+
+        it("lets the owner to begin the process", async () => {
+            await oracle.beginGasPriceUpdate(123, {from: owner})
+            // ok, no revert
+        })
+
+        it("does not allow to finalize change without initiating it first", async () => {
+            await expectRevert(
+                oracle.finalizeGasPriceUpdate(),
+                "Change not initiated"
+            )
+        })
+
+        it("does not allow to finalize change before governance delay", async () => {
+            await oracle.beginGasPriceUpdate(123, {from: owner})
+            await time.increase(time.duration.minutes(359)) // 5 hours and 59 minutes
+            await expectRevert(
+                oracle.finalizeGasPriceUpdate(),
+                "Governance delay has not elapsed"
+            )
+        })
+
+        it("updates value when finalizing the change", async () => {
+            const newValue = 9129111
+            await oracle.beginGasPriceUpdate(newValue, {from: owner})
+            await time.increase(time.duration.minutes(361)) // 6 hours and 1 minute
+            await oracle.finalizeGasPriceUpdate()
+
+            expect(await oracle.gasPrice()).to.eq.BN(newValue)
+        })
+
+        it("does not allow to finalize the change twice", async () => {
+            const newValue = 1111
+            await oracle.beginGasPriceUpdate(newValue, {from: owner})
+            await time.increase(time.duration.minutes(361)) // 6 hours and 1 minute
+            await oracle.finalizeGasPriceUpdate()   
+            await expectRevert(
+                oracle.finalizeGasPriceUpdate(),
+                "Change not initiated"
+            )
+        })
+
+        it("emits an event when finalizing the change", async () => {
+            const newValue = web3.utils.toBN(55555)
+            await oracle.beginGasPriceUpdate(newValue, {from: owner})
+            await time.increase(time.duration.minutes(361)) // 6 hours and 1 minute
+            const receipt = await oracle.finalizeGasPriceUpdate()
+            await expectEvent(receipt, "GasPriceUpdated", {
+                newValue: newValue
+            })
+        })
+    
+        it("notifies consumer contracts when finalizing the change", async () => {
```

Comment:
Can we add a case where we call `beginGasPriceUpdate` just before `governanceDelay` passes so we can check that the new value gets overwritten and the timer is set up again?

Reply:
Done in https://github.com/keep-network/keep-core/pull/1991/commits/cc36463d1c83549fd4584481aec257f505af949b